File: RQ1/Mixed-Manual/task42.c
void closest_integer(const char* value, int len) {
    assert(len > 0);  // PRE
    if (*strchr(value, '.') != '\0') {
        // remove trailing zeros
        while (value[len - 1] == '0')
            --len;
    }
    float num = atof(value);
    float res;
    if (value[len - 2] == '.' && value[len - 1] == '5') {
        if (num > 0)
            res = ceil(num);
        else
            res = floor(num);
    } else if (len > 1 || value[0] != '0')
        res = int(round(num));
    else
        res = 0;

    assert(abs(res) <= abs(atof(value)));  // POST
}

File: RQ1/Mixed-Manual/task42.py
def closest_integer(value):
    assert len(value) > 0  # PRE
    if value.count('.') == 1:
        # remove trailing zeros
        while value[-1] == '0':
            value = value[:-1]

    num = float(value)

    if value[-2:] == '.5':
        if num > 0:
            res = ceil(num)
        else:
            res = floor(num)
    elif len(value) > 1 or value[0] != '0':
        res = int(round(num))
    else:
        res = 0

    assert abs(res) <= abs(float(value))  # POST
